src/index.tsx

```diff
@@ -86,6 +96,80 @@ const rrfProps = {
   createFirestoreInstance,
 };
 
+if (BUGSNAG_API_KEY) {
+  const DEVELOPMENT = "development";
+  const TEST = "test";
+  const STAGING = "staging";
+  const PRODUCTION = "production";
+  const SPARKLEVERSE = "sparkleverse";
+
+  const releaseStage = () => {
+    if (
+      window.location.host.includes("localhost") ||
+      process.env.NODE_ENV === DEVELOPMENT
+    ) {
+      return DEVELOPMENT;
+    }
+
+    if (process.env.NODE_ENV === TEST) {
+      return TEST;
+    }
+
+    if (
+      window.location.host.includes(STAGING) ||
+      BUILD_BRANCH?.includes(STAGING)
+    ) {
+      return STAGING;
+    }
+
+    if (BUILD_BRANCH?.includes("master")) {
+      return PRODUCTION;
+    }
+
+    if (BUILD_BRANCH?.includes(SPARKLEVERSE)) {
+      return SPARKLEVERSE;
+    }
+
+    return process.env.NODE_ENV;
+  };
+
+  Bugsnag.start({
+    apiKey: BUGSNAG_API_KEY,
+    plugins: [new BugsnagPluginReact()],
+    appType: "client",
+    appVersion: BUILD_SHA1,
+    enabledReleaseStages: [STAGING, PRODUCTION, SPARKLEVERSE], // don't track errors in development/test
+    releaseStage: releaseStage(),
+    maxEvents: 25,
+    metadata: {
+      BUILD_SHA1,
+      BUILD_TAG,
+      BUILD_BRANCH,
+      BUILD_PULL_REQUESTS,
+    },
+    onError: (event) => {
+      const { currentUser } = firebase.auth();
+
+      if (!currentUser) return;
+
+      // Add user context to help locate related errors for support
+      event.setUser(
+        currentUser.uid,
+        currentUser.email || undefined,
+        currentUser.displayName || undefined
+      );
+    },
+  });
+}
+
+const BugsnagErrorBoundary = Bugsnag?.getPlugin("react")?.createErrorBoundary(
```

Comment:
Can we nest error boundaries?

Reply:
Just thinking we'll need smaller boundaries if we go with your plan to have boundaries throughout the app.

Reply:
My understanding is that we can have as many as we need within our component tree. They act kind of like a `try { } catch { }` around a tree of components, and stop the error bubbling any higher (eg. to break the entire app)

- https://reactjs.org/docs/error-boundaries.html

If we introduce more at a deeper level (eg. if we wanted to prevent issues in a certain component/feature from being able to bring down the entire app) we would need to look closer at what additional we need to do in them to ensure they still get reported to bugsnag. It may be as simple as using this same `Bugsnag?.getPlugin("react")?.createErrorBoundary` pattern to create the 'deeper' error boundary. Alternatively if it was just a standard boundary, we can manually pass errors through to Bugsnag from it with `Bugsnag.notify` (which we can also use anywhere we want to explicitly pass an already 'handled' error through, eg. instead of `console.error` logging a failed ajax request/similar.

Reply:
Sounds good.